dts: arm: st: l4: Move CAN1 and TIM7 node

gudnimg opened the pull request:
The device tree includes CAN node can1 for all
STM32L4 chips when in fact they don't all support it.
This affects STM32L412xx and STM32L422xx.
Fixes #33896

This PR also moves: 
* TIM7 peripheral node for the same reason as CAN1 for STM32L4.
* USART1 peripheral for STM32L0 since STM32L011 and STM32L031 do not support it.

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

gudnimg committed:
dts: arm: st: l4: Move can1 node

The device tree includes CAN node can1 for all
STM32L4 chips when in fact they don't all support it.
This affects STM32L412xx and STM32L422xx.
Fixes #33896

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

GunZi200 commented:
* Fixed a trailing white space, which failed CI.
* I also noticed just now that Timer 7 is added incorrectly like node for CAN1. Will fix that! Marking this as a draft meanwhile.

gudnimg committed:
dts: arm: st: l4: Move timers7 node

TIM7 peripheral was defined for all STM32L4 chips
in the device tree when not every chip supports it.
This commit moves the node so that only chips
which support the peripheral have the node defined.

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

gudnimg committed:
dts: arm: st: l0: Move node USART1

STM32L011 and STM32L031 do not support USART1.
This commit moves the definition of the node to only
be present for MCUs which support the peripheral.

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

FRASTM reviewed (approved):
stm32l475 and stm32l476 also have CAN1
like the stm32L4R5, the STM32L4S5  also has CAN1

GunZi200 commented:
> stm32l475 and stm32l476 also have CAN1
> like the stm32L4R5, the STM32L4S5 also has CAN1

Yes I tried to be careful to not repeat the nodes and use the `include` where possible.

* stm32l475  and stm32l476  get their CAN1 node from `dts/arm/st/l4/stm32l471.dtsi` via include
* STM32L4S5 gets its CAN1 node from `dts/st/l4/stm32l4r5.dtsi` via include
